src/braket/pennylane_plugin/translation.py

```diff
@@ -57,19 +40,177 @@ def supported_operations() -> FrozenSet[str]:
     Returns:
         FrozenSet[str]: The names of the supported operations
     """
-    return frozenset(_OPERATION_MAP.keys())
+    return frozenset(op.__name__ for op in translate_operation.registry)
 
 
-def translate_operation(operation: Operation) -> Type[Gate]:
-    """Translates a PennyLane ``Operation`` into the corresponding Braket gate.
+@singledispatch
+def translate_operation(operation: Operation, parameters) -> Gate:
+    """Translates a PennyLane ``Operation`` into the corresponding Braket ``Gate``.
 
     Args:
-        operation (Operation): The Pennylane ``Operation`` to translate
+        operation (Operation): The PennyLane ``Operation`` to translate
+        parameters: The parameters of the operation
 
     Returns:
-        Type[Gate]: The `Gate` class corresponding to the given operation
+        Gate: The Braket gate corresponding to the given operation
     """
-    return _OPERATION_MAP[operation.name]
+    raise NotImplementedError(
```

Comment:
Also add a unit test that calls `translate_operation` on an unsupported class just for completeness

Reply:
this already exists-- test_braket_device.test_apply_unsupported

Reply:
Weird; for some reason Codecov's complaining. Not a problem then.

Reply:
Weird; for some reason Codecov was complaining. Not a problem then.